name-clusters/script.whizzml

```diff
@@ -0,0 +1,121 @@
+  (let (model (fetch model-id)
+        importances-ids (model ["model" "importance"]))
+    (map (lambda (n) (head n))
+         importances-ids)))
+
+
+;; generate-models
+;;
+;; Returns a list of the model IDs for the clusters when finished
+;;
+;; Inputs:
+;;   cluster-id: (string) Cluster ID
+;;   centroid-ids: (list) List of centroid IDs
+;;
+;; Output: (list) String representing the field = value name
+(define (generate-models cluster-id centroid-ids)
+  (let (cluster (fetch cluster-id)
+        models (cluster "cluster_models" {})
+        model-ids (for (centroid-id centroid-ids)
+                    (if (models centroid-id false)
+                        (str "model/" (models centroid-id))
+                        (create "model" {"cluster" cluster-id
+                                         "centroid" centroid-id}))))
+    (wait* model-ids)))
+
+
+;; cluster-name
+;;
+;; Creates cluster name  using the most important field names of its
+;; correspondent model and the centroid values
+;;
+;; Inputs:
+;;   clusters: (map) Structure of changes to set the names
+;;   centroid: (map) Centroid info
+;;   fields: (map) Cluster fields
+;;   model-id: (map) Model ID
+;;   number-of-terms: (number) Number of terms to use in cluster name
+;;   separator: (string) Separator to join terms to form cluster names
+;;
+;; Outputs: (string) Cluster name
+;;
+(define (cluster-name clusters
+                      centroid
+                      fields
+                      model-id
+                      number-of-terms
+                      separator)
+  (let (importances-ids (model-importances model-id)
+        name-strs (map (lambda (x) (str (fields [x "name"] "")
+                                        " = "
+                                        (centroid ["center" x])))
+                       importances-ids)
+        n (- number-of-terms 1)
+    ;; List of already named clusters to avoid repeating names
+        seen (map (lambda (c) (c "name")) (values clusters)))
+    (loop (cs (tail name-strs))
+     (if (empty? cs)
+        (head name-strs)
+        (let (c (join separator (cons (head name-strs) (take n cs))))
+          (if (member? c seen)
+        (recur (tail cs))
+        c))))))
+
+
+;; rename-clusters
+;;
+;; Retrieves a cluster and renames it using the most important
+;; field names of its correspondent model
+;;
+;; Inputs:
+;;   cl-id: (resource-id) ID of the cluster model
+;;   number-of-terms: (number) Number of terms to use in cluster name
+;;   separator: (string) Separator to join terms to form cluster names
+;;
+;; Output: (list) List of cluster names
+;;
+(define (rename-clusters cl-id number-of-terms separator)
+  (let (cl (fetch cl-id)
+        fields (cl ["clusters" "fields"])
+        models? (cl "model_clusters")
+        centroids (cl ["clusters" "clusters"])
+        centroid-ids (map (lambda (c) (c "id")) centroids)
+        ;; generate models for all centroids
+        model-ids (when models?
```

Comment:
instead of logging an error below and fail silently, i'd just raise an exception:

```
_ (or (cl "model_clusters") (raise "Cluster doesn't have any model"))
```

and then you can forget about `models?` and all the logic checking it.